spec/index.bs

```diff
@@ -2786,6 +2786,32 @@ interface GPUCommandEncoder {
 GPUCommandEncoder includes GPUObjectBase;
 </script>
 
+{{GPUCommandEncoder}} has the following internal slots:
+
+<dl dfn-type=attribute dfn-for="GPUCommandEncoder">
+    : <dfn>\[[state]]</dfn> of type [=encoder state=].
+    ::
+        The current state of the {{GPUCommandEncoder}}, initially set to [=encoder state/open=].
+</dl>
+
+Each {{GPUCommandEncoder}} has a current <dfn>encoder state</dfn> on the [=Content timeline=]
+which may be one of the following:
+
+ - "<dfn for="encoder state">open</dfn>" where the {{GPUCommandEncoder}} is available to begin new operations.
+ - "<dfn for="encoder state">recording a render pass</dfn>" where the {{GPUCommandEncoder}} has an active {{GPURenderPassEncoder}}.
+ - "<dfn for="encoder state">recording a compute pass</dfn>" where the {{GPUCommandEncoder}} has an active {{GPUComputePassEncoder}}.
+ - "<dfn for="encoder state">closed</dfn>" where the {{GPUCommandEncoder}} is
+     no longer available for any operations.
+
+<div class=note>
+    {{GPUCommandEncoder}} has a state machine where the states are:
+
+     - [=encoder state/open=] when initially created until {{GPUComputePassEncoder/finish()}} is called and when the {{GPUCommandEncoder}} has no active {{GPURenderPassEncoder}} or {{GPUComputePassEncoder}}.
```

Comment:
It seems weird that this is effectively describing an `enum` without saying that this is an enum. I wonder if we can declare this as a real internal enum type instead?

Reply:
I'm not sure if bikeshed will allow that without it becoming an part of the IDL, which we definitely don't want. I can give it a try, though. For what it's worth this particular syntax was lifted from the `buffer state` definition elsewhere in the file, so if we land on a better way of declaring it we should update that one as well.

Reply:
ok, no worries. We'll see if we can refactor this as a follow-up.

Reply:
Turns out we CAN declare these to be an enum and enum values. It doesn't really change much about how the rendered spec displays it (italic bold rather than just bold) and it doesn't cause the values to be included in the IDL. It does require you to use the `{{enum/value}}` syntax rather than the `[=enum\value=]` syntax, which I think is mostly a matter of taste. I've updated the PR to reflect it so you can see how it looks.

Reply:
IDL-style enum works for me